stable_baselines3/a2c/a2c.py

```diff
@@ -118,6 +119,19 @@ def __init__(
             ),
         )
 
+        try:
+            if isinstance(device, str) and device != "cpu" and isinstance(policy, str) and policy != "CnnPolicy":
+                with warnings.catch_warnings():
```

Comment:
why do you need to catch the warning?

Also, please make it a helper function, so it can be re-used by PPO and others (for instance, in the SB3 contrib repo).

The condition should be `policy == "MlpPolicy"` and the device object should be a pytorch device one (so using `self.device` if I recall).